src/debuggerCommunicationServer.ts

```diff
@@ -6,15 +6,32 @@ import * as socketio from "socket.io";
 import { WebviewPanel } from "vscode";
 import { SERVER_INFO } from "./constants";
 
+const DEBUGGER_MESSAGES = {
+    EMITTER: {
+        INPUT_CHANGED: "input_changed",
+        RECEIVED_STATE: "received_state",
+        DISCONNECT: "frontend_disconnected",
+    },
+    LISTENER: {
+        UPDATE_STATE: "updateState",
+        RECEIVED_STATE: "receivedState",
+        DISCONNECT: "disconnect",
+    },
+};
+
 export class DebuggerCommunicationServer {
     private port: number;
     private serverHttp: http.Server;
     private serverIo: socketio.Server;
     private simulatorWebview: WebviewPanel | undefined;
+    private currentActiveDevice;
+    private isWaitingResponse = false;
```

Comment:
nit: similar to the comment below, we can change this to `isPendingResponse`? It's up to you though